Database/PostgreSQL/Simple/Streaming.hs

```diff
@@ -474,8 +475,10 @@ liftMask maskVariant k = do
         unmask (Return q) = Return q
         unmask (Effect m) = Effect $ do
             -- retrieve base's unmask and apply to merged action
-            Masked unmaskVariant <- liftIO $ readIORef ioref
-            unmaskVariant (m >>= chunk >>= return . unmask)
+            liftIO (readIORef ioref) >>= \case
+              Unmasked -> m
+              Masked unmaskVariant ->
+                unmaskVariant (m >>= chunk >>= return . unmask)
```

Comment:
I think that the invariant here is that if `unmask` is ever called, then `ioref` already contains the relevant `Masked unmaskVariant`. So if  we ever get `Unmasked`, it is a runtime-error, which in the original code would be signalled by the pattern-match error. To preserve the original behaviour, I'd insert an `error` call here, e.g.:

```haskell
liftIO (readIORef ioref) >>= \case
  Unmasked ->
    error "unmask called without having masked"
  Masked unmaskVariant ->
    unmaskVariant (m >>= chunk >>= return . unmask)
```